### PR TITLE
fix: duplicate page name error does not disappear on navigating to other pages

### LINKED ISSUE #35949: [Bug]: Duplicate page name error does not disappear on navigating to other pages
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Error on setting duplicate page name does not disappear when we got to some other page's settings.

### Steps To Reproduce

1. Create an application & add 3 pages (Page1, Page2, Page3).
2. Go to Page2 settings & rename it to Page1. This will throw an error as expected.
3. Now go to Page3 settings.

**Expected:** The error should disappear when navigating to a different page's settings.

### Public Sample App

_No response_

### Environment

Production

### Severity

Medium (Frustrating UX)

### Issue video log

https://jam.dev/c/e28bed4e-c5a4-4e2c-8326-ebccec882ea6

### Version

Cloud

### PR DESCRIPTION
**Pull request by a6hishekpandey**
## Description 
**Issue:** Duplicate page name error does not disappear on navigating to other pages.

This PR addresses issue #35949 by resetting the isPageNameValid state to undefined using useEffect hook when the Page settings loads.

Fixes #35949 

> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags=""

### :mag: Cypress test results

> [!CAUTION]  
> If you modify the content in this section, you are likely to disrupt the CI result for your PR.

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved page name validation by resetting its state when the page data updates, ensuring accurate checks based on new information.
- **Style**
	- Minor formatting adjustment for better code readability.

### CODE REVIEW

File: app/client/src/pages/Editor/AppSettingsPane/AppSettings/PageSettings.tsx [hunk 1]
```diff
@@ -126,6 +126,10 @@ function PageSettings(props: { page: Page }) {
     }
   }, [isUpdatingEntity]);
 
+  useEffect(() => {
+    setIsPageNameValid(undefined);
```
Comment (alex-golovanov) [inline app/client/src/pages/Editor/AppSettingsPane/AppSettings/PageSettings.tsx:130]: Hey @a6hishekpandey, thanks for contributing! I see a problem with `isPageNameValid` in that it's being used inappropriately - judging by the name it should be a boolean, but being assigned `string, undefined & boolean` interchangeably. I think we should either have two separate variables `isPageNameValid (boolean) & pageNameError (string | null)` or a single `pageNameError (string | null)`. Either of those is more appropriately typed and avoids assignment of `undefined`.

What do you think?

PS - I understand that this has been composed in such way before you, but we'd like to clean up bits like this as we go forward.

Comment (alex-golovanov) [reply app/client/src/pages/Editor/AppSettingsPane/AppSettings/PageSettings.tsx]: @a6hishekpandey  It seems that there is a problem with formatting - `prettier` check is failing, additionally there may be an issue with `eslint` check. Could you please run/fix those locally, those checks must pass in order for PR to be merged.

Comment (a6hishekpandey) [reply app/client/src/pages/Editor/AppSettingsPane/AppSettings/PageSettings.tsx]: Hey @alex-golovanov addressed prettier's style issue.

**Comment by alex-golovanov** [comment]
@a6hishekpandey everything seems to be in order, we just need to run the test suite on a shadow PR, however some tests are failing for reasons unrelated to your changes, so we need to wait until the issue is resolved. Just keeping you updated.

**Comment by a6hishekpandey** [comment]
@alex-golovanov okay, thanks for the update.

**Comment by a6hishekpandey** [comment]
@alex-golovanov can you please check now if PR is ready for approval or not?

**Comment by alex-golovanov** [comment]
> @alex-golovanov can you please check now if PR is ready for approval or not?
> 
> 

The situation with failing tests has not been resolved yet, I'm keeping an eye on it, we'll merge this once everything is in order.

**Comment by a6hishekpandey** [comment]
okay thankyou, I have 2 more open PRs if you can help me with the review of those as well it will be helpful, linking here https://github.com/appsmithorg/appsmith/pull/36378 and https://github.com/appsmithorg/appsmith/pull/36367.

**Comment by alex-golovanov** [comment]
All tests are passing in [the shadow PR](https://github.com/appsmithorg/appsmith/pull/36381).

**Review by alex-golovanov — APPROVED** [review]

**Pull request merged**